src/transformers/modeling_utils.py

```diff
@@ -2575,6 +2603,20 @@ def save_pretrained(
 
         # Save the model
         for shard_file, shard in shards.items():
+            # remake shard with onloaded parameters if necessary
+            if module_map:
+                # init state_dict for this shard
+                state_dict = {name: "" for name in shard}
+                for module_name in state_dict.keys():
+                    module = module_map[module_name]
+                    # update state dict with onloaded parameters
+                    state_dict = get_state_dict_from_offload(module, module_name, state_dict)
+
+                # assign shard to be the completed state dict
+                shard = state_dict
+                del state_dict
+                gc.collect()
+
```

Comment:
Could you add a check to see if the users indeed have the latest version of accelerate ? See an similar example [here](https://github.com/huggingface/transformers/pull/31052/files#:~:text=if%20quanto_version%20%3E%3D,qint2%2C%20qint4).